itests/kit2/node_full.go

```diff
@@ -33,3 +36,50 @@ func (f *TestFullNode) CreateImportFile(ctx context.Context, rseed int, size int
 	require.NoError(f.t, err)
 	return res, path
 }
+
+// WaitTillChain waits until a specified chain condition is met. It returns
+// the first tipset where the condition is met.
+func (f *TestFullNode) WaitTillChain(ctx context.Context, pred ChainPredicate) *types.TipSet {
```

Comment:
I introduced this method, and generalised it with a little bit of functional magic. I was also able to simplify the deals itest.